### DISCUSSION TITLE
Can we incorporate an API documentation analysis plugin?

### DISCUSSION THREAD

**Discussion by qcgm1978** [danny-avila/LibreChat#293] [Feature Requests & Suggestions]
I came across a project called https://github.com/egeucak/api-doc-gpt, which is an AI tool that assists in discovering API documentation. Is it possible to integrate this tool into our project as a plugin instead of hard coding it?
<img width="933" alt="Screenshot 2023-05-18 at 02 02 32" src="https://github.com/danny-avila/chatgpt-clone/assets/3024299/b8c2ab22-dd52-428f-b28d-aa27cd98f866">
<img width="933" alt="Screenshot 2023-05-18 at 02 05 00" src="https://github.com/danny-avila/chatgpt-clone/assets/3024299/664d0609-0b73-45e5-8039-719c40f46917">

**Comment by danny-avila** [comment]
Currently, we are working on a plugins features where you can use tools like this. That sounds like a great idea for a tool. In the current state, all plugins/tools need to be "hardcoded" in the sense they are langchain tools. It looks like you already did a lot of work integrating it, you should consider it making it a langchain tool so that it can be used by the upcoming "plugins" feature: https://js.langchain.com/docs/modules/agents/tools/

**Comment by qcgm1978** [comment]
I’ll study it. I don’t know there’s so much knowledge about language model plugin system.
